web/src/lib/components/user-settings-page/partner-selection-modal.svelte

```diff
@@ -0,0 +1,84 @@
+<script lang="ts">
+	import { api, UserResponseDto } from '@api';
+	import BaseModal from '../shared-components/base-modal.svelte';
+	import CircleAvatar from '../shared-components/circle-avatar.svelte';
+	import ImmichLogo from '../shared-components/immich-logo.svelte';
+	import Button from '../elements/buttons/button.svelte';
+	import { createEventDispatcher, onMount } from 'svelte';
+
+	let availableUsers: UserResponseDto[] = [];
+	let selectedUsers: UserResponseDto[] = [];
+
+	const dispatch = createEventDispatcher();
```

Comment:
Should include types for all events that can be dispatched, so in this component that would be for `close` and `add-user`